### PR TITLE
[Merged by Bors] - chore: remove Complex/RCLike bit0/1 lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
These were interesting when they worked on number literals, which is no longer the case in lean4.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/IsROrC/Basic.lean [hunk 1]
```diff
@@ -368,17 +341,8 @@ theorem conj_ofReal (r : ℝ) : conj (r : K) = (r : K) := by
...
-  map_bit0 _ _
-#align is_R_or_C.conj_bit0 IsROrC.conj_bit0
-
-set_option linter.deprecated false in
-@[deprecated, isROrC_simps] -- porting note (#10618): was `simp` but `simp` can prove it
-theorem conj_bit1 (z : K) : conj (bit1 z) = bit1 (conj z) :=
-  map_bit1 _ _
-#align is_R_or_C.conj_bit1 IsROrC.conj_bit1
+#noalign is_R_or_C.conj_bit0
+#noalign is_R_or_C.conj_bit1
```
Comment (eric-wieser) [inline Mathlib/Data/IsROrC/Basic.lean:345]: Can you add `conj_ofNat` to replace these?

**Review by eric-wieser — COMMENTED** [review]
I think every bit0/bit1 pair should be replaced by an `ofNat` lemma

**Comment by Ruben-VandeVelde** [comment]
> I think every bit0/bit1 pair should be replaced by an `ofNat` lemma

These already existed except for conj (though `map_*` also applies there). I'm not sure that makes sense for `I_{z,}pow`, but maybe I should restore those for $2n$ and $2n+1$ instead.

File: Mathlib/Data/Complex/Basic.lean [hunk 2]
```diff
@@ -201,44 +200,18 @@ theorem add_im (z w : ℂ) : (z + w).im = z.im + w.im :=
...
-#align complex.bit0_im Complex.bit0_im
-
-@[simp]
-theorem bit1_im (z : ℂ) : (bit1 z).im = bit0 z.im :=
-  add_zero _
-#align complex.bit1_im Complex.bit1_im
+#noalign complex.bit0_re
+#noalign complex.bit1_re
+#noalign complex.bit0_im
+#noalign complex.bit1_im
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Basic.lean:206]:
```suggestion
-- replaced by `re_ofNat`
#noalign complex.bit0_re
#noalign complex.bit1_re
-- replaced by `im_ofNat`
#noalign complex.bit0_im
#noalign complex.bit1_im
```

File: Mathlib/Data/Complex/Basic.lean [hunk 3]
```diff
@@ -201,44 +200,18 @@ theorem add_im (z w : ℂ) : (z + w).im = z.im + w.im :=
...
-#align complex.of_real_bit0 Complex.ofReal_bit0
-
-@[simp, norm_cast]
-theorem ofReal_bit1 (r : ℝ) : ((bit1 r : ℝ) : ℂ) = bit1 (r : ℂ) :=
-  ext_iff.2 <| by simp [bit1]
-#align complex.of_real_bit1 Complex.ofReal_bit1
-
-end
+#noalign complex.of_real_bit0
+#noalign complex.of_real_bit1
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Basic.lean:214]:
```suggestion
-- replaced by `ofReal_ofNat`
#noalign complex.of_real_bit0
#noalign complex.of_real_bit1
```

File: Mathlib/Data/IsROrC/Basic.lean [hunk 4]
```diff
@@ -368,17 +341,14 @@ theorem conj_ofReal (r : ℝ) : conj (r : K) = (r : K) := by
...
```
Comment (eric-wieser) [inline Mathlib/Data/IsROrC/Basic.lean:345]:
```suggestion
-- replaced by `IsROrC.conj_ofNat`
#noalign is_R_or_C.conj_bit0
#noalign is_R_or_C.conj_bit1
```

File: Mathlib/Data/IsROrC/Basic.lean [hunk 5]
```diff
@@ -216,17 +198,8 @@ theorem ofReal_add (r s : ℝ) : ((r + s : ℝ) : K) = r + s :=
...
-  ofReal_add _ _
-#align is_R_or_C.of_real_bit0 IsROrC.ofReal_bit0
-
-set_option linter.deprecated false in
-@[deprecated, simp, isROrC_simps, norm_cast]
-theorem ofReal_bit1 (r : ℝ) : ((bit1 r : ℝ) : K) = bit1 (r : K) :=
-  map_bit1 (algebraMap ℝ K) r
-#align is_R_or_C.of_real_bit1 IsROrC.ofReal_bit1
+#noalign is_R_or_C.of_real_bit0
+#noalign is_R_or_C.of_real_bit1
```
Comment (eric-wieser) [inline Mathlib/Data/IsROrC/Basic.lean:202]:
```suggestion
-- replaced by `IsROrC.ofReal_ofNat`
#noalign is_R_or_C.of_real_bit0
#noalign is_R_or_C.of_real_bit1
```

File: Mathlib/Data/IsROrC/Basic.lean [hunk 6]
```diff
@@ -180,28 +180,10 @@ theorem ofReal_inj {z w : ℝ} : (z : K) = (w : K) ↔ z = w :=
...
-
-set_option linter.deprecated false in
-@[deprecated, simp, isROrC_simps]
-theorem bit1_im (z : K) : im (bit1 z) = bit0 (im z) := by
-  simp only [bit1, map_add, bit0_im, one_im, add_zero]
-#align is_R_or_C.bit1_im IsROrC.bit1_im
+#noalign is_R_or_C.bit0_re
+#noalign is_R_or_C.bit1_re
+#noalign is_R_or_C.bit0_im
+#noalign is_R_or_C.bit1_im
```
Comment (eric-wieser) [inline Mathlib/Data/IsROrC/Basic.lean:186]: Do we have analogs for these?

File: Mathlib/Data/Complex/Basic.lean [hunk 7]
```diff
@@ -490,17 +463,8 @@ theorem coe_imAddGroupHom : (imAddGroupHom : ℂ → ℝ) = im :=
...
-set_option linter.deprecated false
-@[simp]
-theorem I_pow_bit0 (n : ℕ) : I ^ bit0 n = (-1 : ℂ) ^ n := by rw [pow_bit0', Complex.I_mul_I]
-set_option linter.uppercaseLean3 false in
-#align complex.I_pow_bit0 Complex.I_pow_bit0
-
-@[simp]
-theorem I_pow_bit1 (n : ℕ) : I ^ bit1 n = (-1 : ℂ) ^ n * I := by rw [pow_bit1', Complex.I_mul_I]
-set_option linter.uppercaseLean3 false in
-#align complex.I_pow_bit1 Complex.I_pow_bit1
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Basic.lean:503]: I think probably we should keep these, and any other lemmas about `pow`/`zpow` and `bit0`/`bit1`, for now. The rest of the patch is good to go modulo some comments.

**Review by eric-wieser — APPROVED** [review]

File: Mathlib/Data/IsROrC/Basic.lean (hunk 6 above)
Comment (Ruben-VandeVelde) [inline Mathlib/Data/IsROrC/Basic.lean:186]:
```suggestion
-- replaced by `IsROrC.ofNat_re`
#noalign is_R_or_C.bit0_re
#noalign is_R_or_C.bit1_re
-- replaced by `IsROrC.ofNat_im`
#noalign is_R_or_C.bit0_im
#noalign is_R_or_C.bit1_im
```

**Comment by dupuisf** [comment]
Can you change `ROrCLike` to `RCLike` in the comments?

**Comment by Ruben-VandeVelde** [comment]
Done, thanks

**Comment by dupuisf** [comment]
Thanks!

bors r+

File: Mathlib/Analysis/RCLike/Basic.lean [hunk 8]
```diff
@@ -180,28 +180,12 @@ theorem ofReal_inj {z w : ℝ} : (z : K) = (w : K) ↔ z = w :=
...
-@[deprecated, simp, rclike_simps]
-theorem bit1_im (z : K) : im (bit1 z) = bit0 (im z) := by
-  simp only [bit1, map_add, bit0_im, one_im, add_zero]
-#align is_R_or_C.bit1_im RCLike.bit1_im
+-- replaced by `RCLike.ofNat_re`
+#noalign is_R_or_C.bit0_re
+#noalign is_R_or_C.bit1_re
+-- replaced by `RCLike.ofNat_im`
+#noalign is_R_or_C.bit0_im
+#noalign is_R_or_C.bit1_im
```
Comment (YaelDillies) [inline Mathlib/Analysis/RCLike/Basic.lean:188]: Why disregard the align mechanism?
```suggestion
#align is_R_or_C.bit0_re RCLike.ofNat_re
#align is_R_or_C.bit1_re RCLike.ofNat_re
#align is_R_or_C.bit0_im RCLike.ofNat_im
#align is_R_or_C.bit1_im RCLike.ofNat_im
```
Same below

**Pull request closed without merging**